py-polars/src/series.rs

```diff
@@ -822,6 +822,9 @@ impl PySeries {
 
     pub fn to_arrow(&mut self) -> PyResult<PyObject> {
         self.rechunk(true);
+        if matches!(self.series.dtype(), DataType::Object(_)) {
```

Comment:
Pyarrow doesn't support objects, but we use an extension array type for objects that is supported by arrow. This support has utility to go round trip via polars object to arrow and back.

This panic is a regression and should be removed.

Reply:
Do you want me to just remove this panic and leave the old like it was before?

Reply:
Yes. :)